express-notes-api/index.js

```diff
@@ -50,11 +51,12 @@ app.delete('/api/notes/:id', (req, res) => {
   } else if (!notesObject.notes[deleteId]) {
     res.status(404).json({ Error: `cannot find note at id ${deleteId}` });
   } else {
-    res.status(204).json({});
     delete notesObject.notes[deleteId];
     fs.writeFile('./data.json', JSON.stringify(notesObject, null, 2) + '\n', err => {
       if (err) {
         throw res.status(500).json({ Error: 'An unexpected error occured' });
+      } else {
```

Comment:
Oh, i almost missed this. Each `err` you deal with needs to be logged with `console.error(err)` and you want to avoid `throw`, because it will crash your server and exit the process.